useful workflows/target_workflow_2/target_workflow_2-test.wdl

```diff
@@ -0,0 +1,413 @@
+        input:
+            input_file_array = input_file_array,
+            input_string  = "hello world",
+    }
+
+    call lofreq_mock as call_4 {
+        input:
+            input_file_1 = input_file,
+            input_file_2 = input_file
+    }
+
+    call alignment_metrics_mock as call_5 {
+        input:
+            input_file_1         = input_file,
+            input_file_2         = input_file,
+            input_file_optional  = input_file,
+            input_int_optional_1 = 3
+    }
+
+    call run_discordance_mock as call_6 {
+        input:
+            input_file_1         = input_file,
+            input_file_2         = input_file,
+            input_string_default = "some string",
+            input_int_default    = input_int_default
+    }
+
+    call plot_coverage_mock as call_7 {
+        input:
+            input_file   = input_file,
+            input_string = "hello world"
+    }
+
+    call refine_assembly_with_aligned_reads_mock as call_8 {
+        input:
+            input_file_1 = input_file,
+            input_file_2 = input_file,
+            input_string = "hello world"
+    }
+
+    scatter(reads_unmapped_bam in input_file_array) {
+        call align_reads_mock as call_9 {
+            input:
+                input_file_1          = input_file,
+                input_file_2          = input_file,
+                input_file_optional   = input_file,
+                input_string_optional = input_string_default,
+                input_int_optional    = input_int_default
+        }
+    }
+
+    call merge_and_reheader_bams_mock as call_10 {
+        input:
+            input_file_array = input_file_array,
+            input_string  = "hello world"
+    }
+
+    call lofreq_mock as call_11 {
+        input:
+            input_file_1 = input_file,
+            input_file_2 = input_file
+    }
+
+    call plot_coverage_mock as call_12 {
+        input:
+            input_file   = input_file,
+            input_string = "hello world"
+    }
+
+    output {
+        File                        output_file_1           = input_file_array[0]
+        File                        output_file_2           = input_file_array[0]
+        Int                         output_int_1            = input_int_default
+        Int                         output_int_2            = input_int_default
+        Int                         output_int_3            = input_int_default
+        Float                       output_float_1          = input_float_default
+        Int                         output_int_4            = input_int_default
+        Int                         output_int_5            = input_int_default
+        
+        Array[Int]                  output_int_array_1      = [input_int_default]
+        Array[Float]                output_float_array      = [input_float_default]
+        Array[Map[String,String]]   array_of_maps_of_str    = [{input_file: input_string_optional}]
```

Comment:
For the sake of testing our handling, it might be nice to have some variety in size for these maps and arrays (but think 0, 1, 2 rather than 100, 10000, 1000000...)